### ISSUE TITLE
Custom message in error dialog

### ISSUE THREAD

**Issue by greenenvy** [ProjectMirador/mirador#3483]
Hello everyone,

Sorry if i'm at the wrong place for putting the issue, but i could not find a more appropriate place for it.

Is there a possibility to display a custom message to the user if some error happens. For example, if he enters a search query of 2 characters, my search service would return a BadRequest(400) with the error message that the query length is less than 3. Is there a way to elegantly show him the message?

For example, is it possible to do that in the ErrorDialog? Instead of the message displayed in the image (type error: Failed to fetch). I'm not sure if i can influence that message written there, and if i can, what should the format of the response be?

[Link to image](https://i.imgur.com/mmOyrqA.png)

It is not needed to be the ErrorDialog, just any way to elegantly show the message to the user.

Thanks in advance.

**Comment by jbaiter** [comment]
I concur, a less technical way to relay errors to users is something that would be a huge boost in usability, not just for the search use case, but more generally. It happens quite often that Manifests 404s, an annotation list causes a 500, etc.
Having human-readable error messages that are more informative than "failed to fetch" would be great, maybe even just in the current `ErrorDialog`, although a message at the site of the interaction (if applicable) that caused the error would probably be better.
@ggeisler has this come up in UX discussions before?

**Comment by ggeisler** [comment]
@jbaiter @greenenvy I agree we should present more user-friendly error messages. From a UX perspective we didn't get that far when the Mirador 3 team was doing the bulk of our design and development work, so no, to my knowledge there's never been any discussion focused on how to provide better error messages. But this definitely seems like a good area to improve.

I'd be happy to help with this as I can find the time. Perhaps one way to proceed would be for any developers who notice error messages that need to be made more user-friendly (such as the examples @greenenvy and @jbaiter have noted above) could, for each distinct case you come across, post (either in this issue or a new one):

-  A screenshot showing the error message in context (by this I mean capturing the entire Mirador window, rather than just the area directly around the message as in @greenenvy's example above -- not being critical it's just that it can be hard for someone who isn't actively working on Mirador to figure out exactly where in the UI the error is occurring without a broader context).
- If the error occurs in an application that is publicly accessible, a link to the application so we can trigger the error condition directly. If this isn't possible (because the application is still under development, for example), if there's a way to trigger the same error in the [Mirador 3 demo app](https://mirador-dev.netlify.app/__tests__/integration/mirador/) (using a specific manifest, for example) an explanation of the steps necessary required to trigger the error would also be great. 

As we gather these examples I can offer some suggestions for more user-friendly error messages and we can also see if there are any larger patterns that might enable us to generalize how error messages can be better presented to the end-user.

**Comment by marlo-longley** [comment]
Still in discovery phase (collecting examples and patterns). @ggeisler proposed cataloging of error scenarios before designing solutions.
One example that has since surfaced:

- https://github.com/ProjectMirador/mirador/issues/4091 





Original user screenshot:

<img width="557" height="399" alt="Image" src="https://github.com/user-attachments/assets/4ed16f75-58cf-4c90-a5cb-fc923d4ee96c" />

Design decision: establish design patterns for different error types (network, parsing, auth, etc.)
